benchmark/benchmarker.h

```diff
@@ -303,27 +303,35 @@ struct benchmarker {
       exit_error(string("Failed to parse ") + filename + " during stage 1: " + pj.get_error_message());
     }
 
-    // Stage 2 (unified machine)
-    event_count stage2_count;
-    if (!stage1_only || stats == NULL) {
-      if (!stage1_only) {
-        collector.start();
-      }
-      result = parser.stage2((const uint8_t *)json.data(), json.size(), pj);
-      if (!stage1_only) {
-        stage2_count = collector.end();
-        stage2 << stage2_count;
-      }
+    // Stage 2 (unified machine) and the rest
 
+    if (stage1_only) {
+      all_stages << stage1_count;
+    } else {
+      event_count stage2_count;
+      collector.start();
+      result = parser.stage2((const uint8_t *)json.data(), json.size(), pj);
+      stage2_count = collector.end();
+      stage2 << stage2_count;
+      // You would think that the entire processing is just stage 1 + stage 2, but
+      // empirically, that's not true! Not even close to be true in some instances.
+      event_count allstages_count;
```

Comment:
Just a comment for future change: I kinda don't want to run stage 1 and 2 AND both, at least by default, given the size of the difference here. We can add flags or whatever later though.

Reply:
> given the size of the difference here

The difference is huge though...? I am getting pretty significant differences in various tests... and on different systems. Did you try it out?

I agree to make this optional, of course.